core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -125,22 +129,30 @@ class TransactionCoordinator(brokerId: Int,
           val createdMetadata = new TransactionMetadata(transactionalId = transactionalId,
             producerId = producerId,
             producerEpoch = RecordBatch.NO_PRODUCER_EPOCH,
+            lastProducerEpoch = RecordBatch.NO_PRODUCER_EPOCH,
```

Comment:
If there is no producerId currently mapped, the KIP suggests we allow the newly generated producerId to be returned. I wonder if it's simpler from an implementation perspective to accept the producerId and epoch from the InitProducerId request instead.

By the way, I think we might need to be a little careful with the result of `putTransactionStateIfNotExists`. Just need to be sure that if two InitProducerId requests come in at once, only one will be accepted.